src/libPMacc/include/math/RungeKutta/RungeKutta4.hpp

```diff
@@ -0,0 +1,74 @@
+/**
+ * 2015 Richard Pausch
+ *
+ * This file is part of libPMacc.
+ *
+ * libPMacc is free software: you can redistribute it and/or modify
+ * it under the terms of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * libPMacc is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with libPMacc.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+namespace PMacc
+{
+namespace math
+{
+
+
+template<typename F, typename V, typename T>
+class RungeKutta4
+  {
+  public:
+    /** Runge Kutta solver 4th order
+     *
+     * Parameter:
+     * f functor with first argument time and second variables
+     * v variables of type V (can be vector type)
+     * t time 
+     * h time step
+     */
+    RungeKutta4 (F& f, V v, T t, T h) : f(f), v(v), t(t), h(h) {}
+  
+    V next()
+    /** Calculate next time step based on the Runge Kutta
+     *  algorithm and return next variable
+     */
+    {
+      // calculate all 4 steps of the Runge Kutta 4th order
+      const V k_1 = f(t, v);
+      const V k_2 = f(t + 0.5*h, v+((0.5*h)*k_1));
```

Comment:
please use the same spacing in the equation `a + b * ( c + a)` (best is to surround  operators with white spaces)